feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go

```diff
@@ -215,3 +216,80 @@ func TestLinecardReboot(t *testing.T) {
 	helpers.ValidateOperStatusUPIntfs(t, dut, intfsOperStatusUPBeforeReboot, 10*time.Minute)
 	// TODO: Check the line card uptime has been reset.
 }
+
+
+//Reboot the fabric component on the DUT.
+func TestFabricReboot(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	if deviations.GNOIFabricComponentRebootUnsupported(dut) {
+		t.Skipf("Skipping test due to deviation deviation_gnoi_fabric_component_reboot_unsupported")
+	}
+
+	const fabricBootTime = 10 * time.Minute
+	fabrics := components.FindComponentsByType(t, dut, fabricType)
+	t.Logf("Found fabric components: %v", fabrics)
+
+	t.Logf("Find a removable fabric component to reboot.")
+	var removableFaric string
+	for _, fabric := range fabrics {
+		t.Logf("Check if %s is removable", fabric)
+		if got := gnmi.Lookup(t, dut, gnmi.OC().Component(fabric).Removable().State()).IsPresent(); !got {
+			t.Logf("Detected non-removable fabric component: %v", fabric)
+			continue
+		}
+		if got := gnmi.Get(t, dut, gnmi.OC().Component(fabric).Removable().State()); got {
```

Comment:
No need to lookup the value multiple times. Just do the lookup once, i.e.

```
removable, ok := gnmi.Lookup(t, dut, gnmi.OC().Component(fabric).Removable().State()).Val()
```

where `ok` is true when the leaf is present, and `removable` gives you the boolean value of the leaf when it is present. Write your logic in terms of those.

Reply:
Look up once, updated the code as below:

```go
	if removable, ok := gnmi.Lookup(t, dut, gnmi.OC().Component(fabric).Removable().State()).Val(); ok {
			if removable {
				t.Logf("Found removable fabric component: %v", fabric)
				removableFabric = fabric
			}else{
				t.Logf("Found non-removable fabric component: %v", fabric)
			}
		}else{
			t.Logf("Detected non-removable fabric component: %v", fabric)
		}
```